src/StoreApi/Utilities/OrderController.php

```diff
@@ -406,6 +410,30 @@ protected function validate_coupon_usage_limit( \WC_Coupon $coupon, \WC_Order $o
 		}
 	}
 
+	/**
+	 * Check there is a shipping method if it requires shipping.
+	 *
+	 * @throws RouteException Exception if invalid data is detected.
+	 * @param boolean $needs_shipping Current order needs shipping.
+	 * @param array   $chosen_shipping_methods Array of shipping methods.
+	 */
+	public function validate_selected_shipping_methods( $needs_shipping, $chosen_shipping_methods = array() ) {
+		if ( ! $needs_shipping || ! is_array( $chosen_shipping_methods ) ) {
+			return;
+		}
+
+		foreach ( $chosen_shipping_methods as $chosen_shipping_method ) {
+			if ( false === $chosen_shipping_method ) {
+				throw new RouteException(
+					'woocommerce_rest_invalid_shipping_option',
+					__( 'Sorry, this order requires a shipping option', 'woo-gutenberg-products-block' ),
```

Comment:
We usually end error messages with a period:
```suggestion
					__( 'Sorry, this order requires a shipping option.', 'woo-gutenberg-products-block' ),
```